src/psyclone/generator.py

```diff
@@ -254,14 +251,42 @@ def generate(filename, api="", kernel_paths=None, script_name=None,
     alg_gen = None
 
     if api == "gocean1.0":
-        # Create language-level PSyIR from the Fortran file
+        # Create language-level PSyIR from the Algorithm file
         reader = FortranReader()
         psyir = reader.psyir_from_file(filename)
 
         # Raise to Algorithm PSyIR
         alg_trans = AlgTrans()
         alg_trans.apply(psyir)
 
+        # For each kernel called from the algorithm layer
+        kernels = {}
+        for invoke in psyir.walk(AlgorithmInvokeCall):
+            kernels[id(invoke)] = {}
```

Comment:
Does each `AlgorithmInvokeCall` have a unique name that we can use as a key instead?

Reply:
In short, no. There is an index but that is currently internal.